### PR TITLE
Add Postgres + Sequelize

### PR DESCRIPTION
**Pull request by gamburgm**
Long-term goal: add a database to back up our data, that'll let us have a stable data store for our entries (Courses, Professors, etc) and lets us more easily serve that data (and centralize our data) when we can.

This PR just sets up the tables and schema we're going to be using. Getting this out quickly so that our Graduate team can start using it to get data from our new Major table.

### CODE REVIEW

File: models/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const db = {};
+
+let sequelize;
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
+}
+
+fs
```
Comment (dajinchu) [inline models/index.js:26]: Is this importing everything in the folder or something? Is this normal sequelize stuff? I feel weird about this kind of thing

**Review by dajinchu — APPROVED** [review]
Looks good for getting the model set up. Do we have a plan for writing tests? Makes sense for none of this to have tests, but eventually we'll want tests that will hit postgres. What's the game plan for that?
Also what's the game plan for connecting elasticsearch? Either way, in the interest of getting this through so that Graduate isn't blocked, I'd give my :+1:

File: models/index.js (hunk 1 above)
Comment (gamburgm) [reply models/index.js:26]: This was auto-generated by Sequelize, the idea is that you can import `db` anywhere and fetch each model from there—in other words, yeah.

**Review by gamburgm — COMMENTED** [review]
> Looks good for getting the model set up. Do we have a plan for writing tests? Makes sense for none of this to have tests, but eventually we'll want tests that will hit postgres. What's the game plan for that?
> Also what's the game plan for connecting elasticsearch? Either way, in the interest of getting this through so that Graduate isn't blocked, I'd give my 👍

Fortunately, Sequelize comes with configuration for a test DB out of the box, so we'll be able to hit that however we want.

Next step is to import all the JSON from the scrapers and then convert the IDs into Postgres hits, goal is to get that done by Thanksgiving.

**Comment by ryanhugh** [comment]
If y'all are keeping track of historical data you could think about these ideas too: 

https://github.com/ryanhugh/searchneu/issues/53
https://github.com/ryanhugh/searchneu/issues/50

🙂

**Comment by ryanhugh** [comment]
> so that our Graduate team can start using it to get data from our new Major table.

Btw, what are the dets on the new Major table? Where do you guys plan on getting info on Majors and Minors (or do you just plan to manually type them in?). Just curious 😛

**Comment by gamburgm** [comment]
> > so that our Graduate team can start using it to get data from our new Major table.
> 
> Btw, what are the dets on the new Major table? Where do you guys plan on getting info on Majors and Minors (or do you just plan to manually type them in?). Just curious 😛

We got some starting data from the Graduate team that we're going to use for now, but going forward we're hoping that either the Graduate team can insert it themselves or there's a more automated process to do it.

**Pull request merged**